File: android/VideoView/app/src/main/java/io/streamroot/dna/samples/videoview/MainActivity.java
package io.streamroot.dna.samples.videoview;

import android.content.Intent;
import android.os.Bundle;
import android.widget.AutoCompleteTextView;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public final class MainActivity extends AppCompatActivity {
    private AutoCompleteTextView streamEditText;

    @Override
    protected void onCreate(Bundle savedInstance) {
        super.onCreate(savedInstance);

        setContentView(R.layout.activity_main);
        streamEditText = findViewById(R.id.streamEditText);

        Button launchButton = findViewById(R.id.launchButton);
        launchButton.setOnClickListener(v -> {
            Intent intent = new Intent(getApplicationContext(), PlayerActivity.class);

            intent.putExtra("streamUrl", streamEditText.getText().toString());

            intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

            getApplicationContext().startActivity(intent);
        });
    }
}

File: android/VideoView/app/src/main/java/io/streamroot/dna/videoview/MainActivity.kt
package io.streamroot.dna.videoview

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.AutoCompleteTextView
import android.widget.Button

class MainActivity : AppCompatActivity() {
    private lateinit var streamEditText: AutoCompleteTextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        streamEditText = findViewById(R.id.streamEditText)

        findViewById<Button>(R.id.launchButton).setOnClickListener {
            val intent = Intent(applicationContext, PlayerActivity::class.java)

            intent.putExtra("streamUrl", streamEditText.text.toString())

            intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);

            applicationContext.startActivity(intent)
        }
    }
}
